### DISCUSSION TITLE
Compute loss of an image not in training set

### DISCUSSION THREAD

**Discussion by thundercarrot** [NVlabs/instant-ngp#863] [Q&A]
I'm looking for a way to determine how well an image not in the training set is modeled by the trained network. I was thinking that I could compute a scalar loss for that image. I see no function that does this directly and was about to roll my own:

* Perform inference on the network given the new image's pose
* Compute loss at each pixel, given new image
* reduce_sum to obtain a scalar loss

Does that sound right? I'm a little at a loss (ugh) as to how to proceed. Is there code already in ngp I can use for the second step?

**Comment by Tom94** [comment]
Hi there, you can use `float loss = trainer->loss(*trainer->forward(1.0f, input, reference));` to get a cpu-side averaged loss.

Alternatively, your suggested implementation would also work, but would require a little bit more code (and be slightly faster due to being able to use `model->inference()` rather than `model->forward()`).
